src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ConsentOrderApprovedDocumentService.java

```diff
@@ -7,32 +7,115 @@
 
 import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.join;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.YES_VALUE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APPLICANT_REPRESENTED;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_ADDRESS_CCD_FIELD;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_FIRST_AND_MIDDLE_NAME_CCD_FIELD;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_LAST_NAME_CCD_FIELD;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_RESP_FIRST_AND_MIDDLE_NAME_CCD_FIELD;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_RESP_LAST_NAME_CCD_FIELD;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.APP_SOLICITOR_ADDRESS_CCD_FIELD;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.CONSENT_ORDER_APPROVED_NOTIFICATION_LETTER;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_REFERENCE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.CommonFunction.addressLineOneAndPostCodeAreBothNotEmpty;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.CommonFunction.nullToEmpty;
 
 @Service
 @Slf4j
 public class ConsentOrderApprovedDocumentService extends AbstractDocumentService {
 
+    private LetterAddressHelper letterAddressHelper;
+
     @Autowired
     public ConsentOrderApprovedDocumentService(DocumentClient documentClient,
                                                DocumentConfiguration config,
-                                               ObjectMapper objectMapper) {
+                                               ObjectMapper objectMapper,
+                                               LetterAddressHelper letterAddressHelper) {
         super(documentClient, config, objectMapper);
+        this.letterAddressHelper = letterAddressHelper;
     }
 
-
     public CaseDocument generateApprovedConsentOrderLetter(CaseDetails caseDetails, String authToken) {
+        log.info(
+                "Generating Approved Consent Order Letter {} from {} for bulk print for case id: {} ",
+                config.getApprovedConsentOrderFileName(),
+                config.getApprovedConsentOrderTemplate(),
+                caseDetails.getId().toString());
         return generateDocument(authToken, caseDetails,
                 config.getApprovedConsentOrderTemplate(),
                 config.getApprovedConsentOrderFileName());
     }
 
+    public CaseDocument generateApprovedConsentOrderNotificationLetter(CaseDetails caseDetails, String authToken) {
```

Comment:
uuuu, can you please split this method into several private ones, so it's not ssoooo long  [and easier to read / understand]